nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -2368,6 +2368,25 @@ private boolean mayBeNullExpr(VisitorState state, ExpressionTree expr) {
     Symbol exprSymbol = ASTHelpers.getSymbol(expr);
     boolean exprMayBeNull;
     switch (expr.getKind()) {
+      case ARRAY_ACCESS:
+        exprMayBeNull = false;
+        // TODO: export to utility method
+        if (config.isJSpecifyMode()) {
+          ArrayAccessTree arrayAccess = (ArrayAccessTree) expr;
+          ExpressionTree arrayExpr = arrayAccess.getExpression();
+          Symbol arraySymbol = ASTHelpers.getSymbol(arrayExpr);
+          if (arraySymbol != null) {
+            for (Attribute.TypeCompound t : arraySymbol.getRawTypeAttributes()) {
+              for (TypeAnnotationPosition.TypePathEntry entry : t.position.location) {
+                if (entry.tag == TypeAnnotationPosition.TypePathEntryKind.ARRAY) {
```

Comment:
I think we need to check whether the annotation is actually `@Nullable`.  This just checks for the presence of _some_ annotation.  And yes this logic should be extracted to a utility method.

Reply:
Addressed!